contracts/incentives/PullRewardsIncentivesController.sol

```diff
@@ -0,0 +1,59 @@
+// SPDX-License-Identifier: agpl-3.0
+pragma solidity 0.7.5;
+pragma experimental ABIEncoderV2;
+
+import {IERC20} from '@aave/aave-stake/contracts/interfaces/IERC20.sol';
+import {SafeERC20} from '@aave/aave-stake/contracts/lib/SafeERC20.sol';
+
+import {BaseIncentivesController} from './base/BaseIncentivesController.sol';
+
+/**
+ * @title PullRewardsIncentivesController
+ * @notice Distributor contract for ERC20 rewards to the Aave protocol participants that pulls ERC20 from external account
+ * @author Aave
+ **/
+contract PullRewardsIncentivesController is
+  BaseIncentivesController
+{
+  using SafeERC20 for IERC20;
+
+  address internal _rewardsVault;
+
+  event RewardsVaultUpdated(address indexed vault);
+  
+  constructor(IERC20 rewardToken, address emissionManager)
+    BaseIncentivesController(rewardToken, emissionManager)
+  {}
+
+  /**
+   * @dev Initialize AaveIncentivesController
+   * @param rewardsVault rewards vault to pull ERC20 funds
+   **/
+  function initialize(address rewardsVault) external initializer {
+    _rewardsVault = rewardsVault;
+    emit RewardsVaultUpdated(_rewardsVault);
+  }
+
+  /**
+   * @dev returns the current rewards vault contract
+   * @return address
+   */
+  function getRewardsVault() external view returns (address) {
+    return _rewardsVault;
+  }
+
+  /**
+   * @dev update the rewards vault address, only allowed by the Rewards admin
+   * @param rewardsVault The address of the rewards vault
+   **/
+  function setRewardsVault(address rewardsVault) external onlyEmissionManager {
```

Comment:
Is there a risk of updating the rewardsVault, but leaving existing token emissions? For token contract updates, would be fine, but if switching tokens could be an issue. Probably can just be handled with clear documentation but can consider include smart contract options.

Potential options
Clearly document that existing emissions will remain for the new token
Clear all emmissions on `setRewardsVault`
Add a parameters to `setRewardsVault` and include a call to update emissions

Reply:
If the EmissionManager changes the rewards vault, it will not change the current token as is inmmutable, just the source where "pull" and transfer the reward token. In my opinion emissions could still be the same and if the "reward vault" has sufficient tokens the incentives will not stop.

The rewards vault can be an EOA or smart contract with the sole purpose of pulling the reward tokens from this contract to the incentivized accounts.

Reply:
Makes sense +1